pyanaconda/modules/payloads/payload/rpm_ostree/installation.py

```diff
@@ -633,6 +642,162 @@ def _set_kargs(self):
+
+        # Bootc requires empty `boot` directory to be present
+        log.debug("Bootc workaround: create bootc required dirs")
+        # Mount /boot partition created by autopart
+        # Get the boot device
+        device_tree = STORAGE.get_proxy(DEVICE_TREE)
+        boot_device_id = device_tree.GetBootDevice()
+        boot_device_data = DeviceData.from_structure(device_tree.GetDeviceData(boot_device_id))
+        safe_exec_program("mount", [boot_device_data.path, self._sysroot + "/boot"])
+        # Make sure the partition is empty
+        safe_exec_program("rm", ["-rf", self._sysroot + "/boot/*"])
+
+        log.debug("Executing bootc install command")
+        safe_exec_program(
+            "bootc",
+            ["install",
+            "to-filesystem",
+            "--stateroot=" + stateroot,
+            "--source-imgref=" + self._data.sourceImgRef,
+            "--target-imgref=" + self._data.targetImgRef,
+            self._sysroot]
+        )
+
+        # Get the deployment path using OSTree API (same approach as SetSystemRootTask)
+        # This needs to be done before unmounting, while the sysroot is still accessible
+        # After bootc install, the ostree repo is in self._sysroot
+        new_root_path = self._get_deployment_path(self._sysroot)
+
+        # After bootc install is completed sysroot is mounted in read only mode
+        # and it points to the base of ostree deployment but not the new true sysroot.
+        # Final steps of Anaconda install expects to find config dirs like `/etc`
+        # in the /mnt/sysroot. We need to fix those mounts.
+
+        # Track which partition is sysroot
+        # Get the root device
+        device_tree = STORAGE.get_proxy(DEVICE_TREE)
+        root_device_id = device_tree.GetRootDevice()
+        root_device_data = DeviceData.from_structure(device_tree.GetDeviceData(root_device_id))
+        sysroot_partition = root_device_data.path
+
+        # Remove existing mounts as they are read only
+        safe_exec_program("umount", ["-l", "/run/bootc/storage"])
+
+        # Mount the partition to physroot so the deployment path is accessible
+        safe_exec_program("mount", [sysroot_partition, self._physroot])
+
+        # Adjust the deployment path: replace /mnt/sysroot with /mnt/sysimage
+        # since we remounted the partition to self._physroot
+        new_root_path = new_root_path.replace(self._sysroot, self._physroot)
+
+        set_system_root(new_root_path)
+
+        # Anaconda is expecting to put some files in new root directory
+        # but after bootc install root is a symlinking to not existing var/roothome
+        os.makedirs(self._sysroot + "/var/roothome", mode=0o755, exist_ok=True)
+        os.makedirs(self._sysroot + "/var/home", mode=0o755, exist_ok=True)
+
+        # Prepare SELinux hooks needed by the `chpasswd` running in chroot
+        # when SELinux is enabled: https://bugzilla.redhat.com/show_bug.cgi?id=1321375
+        proc_path = "/proc"
```

Comment:
Here we bind mount host's /proc path. I dont think we want that rhought,we probably want the proc  and sys as prepated by blivet's FSSET, the one that we wiped out.